src/utils.rs

```diff
@@ -0,0 +1,114 @@
+use blake2::digest::{Update, VariableOutput};
+use blake2::Blake2bVar;
+use std::cmp::min;
+
+// Oracles
+
+/// Takes as input a hash and inverse probability $np$ and returns a
+/// uniformly distributed integer in [1, np] (c.f. Appendix B, Alba paper).
+pub fn oracle_uniform(hash: &[u8], n: usize) -> usize {
```

Comment:
"oracle" doesn't seem accurate since this function doesn't produce randomness. I would maybe call it "sample"

Reply:
Also, with the risk of being pedantic :), I'd like to see a slightly more clear comment - we don't need to say that n_p is an inverse probability, but we should maybe state how we use the hash.